dipy/viz/ui.py

```diff
@@ -3300,3 +3303,226 @@ def left_button_clicked(self, i_ren, obj, list_box_item):
+        List of extensions to be shown as files.
+    listbox : :class: 'ListBox2D'
+        Container for the menu.
+    """
+
+    def __init__(self, directory_path, extensions=["*"], position=(0, 0),
+                 size=(100, 300), multiselection=True, reverse_scrolling=False,
+                 font_size=20, line_spacing=1.4):
+        """
+        Parameters
+        ----------
+        extensions: list(string)
+            List of extensions to be shown as files.
+        directory_path: string
+            Path of the directory where this dialog should open.
+        position : (float, float)
+            Absolute coordinates (x, y) of the lower-left corner of this
+            UI component.
+        size : (int, int)
+            Width and height in pixels of this UI component.
+        multiselection: {True, False}
+            Whether multiple values can be selected at once.
+        reverse_scrolling: {True, False}
+            If True, scrolling up will move the list of files down.
+        font_size: int
+            The font size in pixels.
+        line_spacing: float
+            Distance between listbox's items in pixels.
+        """
+        self.font_size = font_size
+        self.multiselection = multiselection
+        self.reverse_scrolling = reverse_scrolling
+        self.line_spacing = line_spacing
+        self.extensions = extensions
+        self.current_directory = directory_path
+        self.menu_size = size
+
+        super(FileMenu2D, self).__init__()
+        self.position = position
+        self.set_slot_colors()
+
+    def _setup(self):
+        """ Setup this UI component.
+        Create the ListBox (Panel2D) filled with empty slots (ListBoxItem2D).
+        """
+        self.directory_contents = self.get_all_file_names()
+        content_names = [x[0] for x in self.directory_contents]
+        self.listbox = ListBox2D(
+            values=content_names, multiselection=self.multiselection,
+            font_size=self.font_size, line_spacing=self.line_spacing,
+            reverse_scrolling=self.reverse_scrolling, size=self.menu_size)
+
+        self.add_callback(self.listbox.up_button.actor, "LeftButtonPressEvent",
+                          self.scroll_callback)
+        self.add_callback(self.listbox.down_button.actor,
+                          "LeftButtonPressEvent", self.scroll_callback)
+
+        # Handle mouse wheel events on the panel.
+        up_event = "MouseWheelForwardEvent"
+        down_event = "MouseWheelBackwardEvent"
+        if self.reverse_scrolling:
+            up_event, down_event = down_event, up_event  # Swap events
+
+        self.add_callback(self.listbox.panel.background.actor, up_event,
+                          self.scroll_callback)
+        self.add_callback(self.listbox.panel.background.actor, down_event,
+                          self.scroll_callback)
+
+        # Handle mouse wheel events on the slots.
+        for slot in self.listbox.slots:
```

Comment:
Still not a fan of reimplementing things that are already in `ListBox2D` (see https://github.com/nipy/dipy/pull/1570#issuecomment-397904982 and https://github.com/nipy/dipy/pull/1570#issuecomment-397906003) but I don't have time to think it through and already block you for long enough.